Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -0,0 +1,463 @@
+                    RemComp<CommandStaffComponent>(mind.OwnedEntity.Value);
+                }
+                EnsureComp<HeadRevolutionaryComponent>(mind.OwnedEntity.Value);
+                EnsureComp<RevolutionaryComponent>(mind.OwnedEntity.Value);
+                if (_inventory.TryGetSlotContainer(mind.OwnedEntity.Value, "back", out var containerSlot, out var slotDefinition))
+                {
+                    var bag = containerSlot.ContainedEntity;
+                    var flash = Spawn("Flash", new EntityCoordinates(mind.OwnedEntity.Value, Vector2.Zero));
+                    var glasses = Spawn("ClothingEyesGlassesSunglasses", new EntityCoordinates(mind.OwnedEntity.Value, Vector2.Zero));
+                    if (bag != null && _storageSystem.CanInsert((EntityUid) bag, flash, out var reason))
+                    {
+                        _storageSystem.Insert((EntityUid) bag, flash);
+                        _storageSystem.Insert((EntityUid) bag, glasses);
+                    }
+                    else
+                    {
+                        //Honestly if they don't have a bag they are just going to take this L.
+                        _entityManager.DeleteEntity(flash);
+                        _entityManager.DeleteEntity(glasses);
+                        _stationSpawningSystem.EquipStartingGear(mind.OwnedEntity.Value, _prototypeManager.Index<StartingGearPrototype>(comp.HeadRevGearPrototypeId), null);
+                    }
+                }
+                _npcFaction.RemoveFaction(mind.OwnedEntity.Value, "NanoTrasen", false);
+                _npcFaction.AddFaction(mind.OwnedEntity.Value, "Revolutionary");
+                inCharacterName = MetaData(mind.OwnedEntity.Value).EntityName;
+                comp.HeadRevs.Add(inCharacterName, headRev.Name);
+                _audioSystem.PlayGlobal(comp.HeadRevStartSound, mind.OwnedEntity.Value);
+            }
+            if (mind.Session != null)
+            {
+                var message = Loc.GetString("head-rev-role-greeting");
+                var wrappedMessage = Loc.GetString("chat-manager-server-wrap-message", ("message", message));
+                _chatManager.ChatMessageToOne(Shared.Chat.ChatChannel.Server, message, wrappedMessage, default, false, mind.Session.ConnectedClient, Color.FromHex("#5e9cff"));
+            }
+        }
+    }
+
+    /// <summary>
+    /// Assigns command staff on station with a component so they can be used later.
+    /// </summary>
+    private void AssignCommandStaff()
+    {
+        var jobs = _prototypeManager.Index<DepartmentPrototype>("Command");
+        var allPlayers = _playerSystem.ServerSessions.ToList();
+        var playerList = new List<IPlayerSession>();
+        foreach (var player in allPlayers)
+        {
+            if (player.AttachedEntity == null || HasComp<HumanoidAppearanceComponent>(player.AttachedEntity))
+            {
+                playerList.Add(player);
+            }
+        }
```

Comment:
Every time any player spawns in you're iterating every single player in the game just to assign a role to one person which is obviously really bad, just check it for that person.